OpenSim/Common/SmoothSegmentedFunction.cpp

```diff
@@ -582,59 +583,100 @@ If x is in the linear region
 ________________________________________________________________________
     */
 
-double SmoothSegmentedFunction::calcDerivative(double x, int order) const
+namespace
+{
+
+// Used to select which derivative orders to compute:
+// e.g. selectedOrders = {true, false, false, true, false, false} selects the
+// zeroth and third order derivative.
+using SelectedDerivativeOrders = std::array<bool, 6>;
```

Comment:
In for a penny, in for a pound - if you're going to use aliases to hide array bounds you mayaswell do it consistently, and with additional compile-time checks, because the code is hard-coding the array bounds in function bodies etc. below

```c++
using SelectedDerivativeOrders = std::array<bool, 6>;
using DerivativeValues = std::array<double, 6>;
static_assert(SelectedDerivativeOrders{}.size() == DerivativeValues.size());
```